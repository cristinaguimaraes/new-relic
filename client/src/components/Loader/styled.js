import styled, { keyframes } from 'styled-components';

const rotate = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
    }
`;

export const Spinner = styled.div`
  opacity: 1;
  position: relative;
  transition: opacity linear 0.1s;
  animation: ${rotate} 2s ease-in-out 0s infinite;

  &::before {
    border: solid 3px #eee;
    border-bottom-color: red;
    border-radius: 50%;
    content: '';
    height: 40px;
    left: 50%;
    opacity: inherit;
    position: absolute;
    top: 50%;
    transform: translate3d(-50%, -50%, 0);
    transform-origin: center;
    width:  40px;
    will-change: transform;
  }
`;

export const Wrapper = styled.div`
padding-top: 50px;
 height: 200px;
 display: flex;
 align-items:center;
 justify-content: center
`
