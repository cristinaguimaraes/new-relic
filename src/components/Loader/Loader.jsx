import React from "react";
import { Spinner, Wrapper } from "./styled";

export const Loader = () => (
  <Wrapper>
    <Spinner />
  </Wrapper>
);
