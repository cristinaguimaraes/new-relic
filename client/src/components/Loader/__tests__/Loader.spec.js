import React from 'react';
import { Loader} from '../Loader';
import { shallow } from 'enzyme';

describe('Loader component', () => {
  test('should render properly', () => {
    expect(shallow(<Loader />)).toMatchSnapshot()
  });
});
