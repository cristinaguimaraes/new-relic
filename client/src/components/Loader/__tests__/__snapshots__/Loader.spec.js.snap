// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should render properly 1`] = `ShallowWrapper {}`;
